src/udev/test-udev-rules-util.c

```diff
@@ -0,0 +1,60 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include "macro.h"
+#include "string-util.h"
+#include "udev-rules-util.h"
+
+
+static void test_parse_value(void) {
+        // input: "valid operand"
+        // parsed: valid operand
+        // use the following command to help generate textual C strings:
+        // python3 -c 'import json; print(json.dumps(input()))'
+        char str[] = "\"valid operand\"";
+        char *expected_endpos = str + strlen(str) - 1;
+        char *value;
+        char *endpos;
+        assert_se(parse_value(str, &value, &endpos) == 0);
+        assert_se(streq_ptr(value, "valid operand"));
+        assert_se(endpos == expected_endpos);
+}
+
+static void test_parse_value_with_backslashes(void) {
+        // input: "va'l\'id\"op\"erand"
+        // parsed: va'l\'id"op"erand
+        char str[] = "\"va'l\\'id\\\"op\\\"erand\"";
+        char *expected_endpos = str + strlen(str) - 1;
+        char *value;
+        char *endpos;
+        assert_se(parse_value(str, &value, &endpos) == 0);
+        assert_se(streq_ptr(value, "va'l\\'id\"op\"erand"));
+        assert_se(endpos == expected_endpos);
+}
+
+static void test_parse_value_no_quotes(void) {
+        char str[] = "no quotes";
+        char *value = (char *) 0x12345678;
+        char *endpos = (char *) 0x87654321;
```

Comment:
this is weird on 32bit systems. Since the constant you specify doesn't fit in "int", it's above INT_MAX. But you use no suffix such as `U`.

I think it would be better to suffix this with `U` at least, to make sure this is an unsigned  literal. maybe even use our UINT_TO_PTR/PTR_TO_UINT macros